charts/skypilot/templates/api-deployment.yaml

```diff
@@ -52,10 +52,16 @@ spec:
           # Any local changes to the config.yaml file will be overwritten by the contents of the configmap.
           cp /tmp/config.yaml /root/.sky/config.yaml
           {{- end }}
-          if sky api start --deploy; then
-            tail -n+0 -f /root/.sky/api_server/server.log
+
+          if sky api start -h | grep -q -- "--foreground"; then
+            exec sky api start --deploy --foreground
```

Comment:
any reason we need to use `exec` instead of directly run `sky api start --deploy --foreground`

Reply:
Add a comment here : )